tools/src/test/resources/dev/run-qualification-tests.sh

```diff
@@ -0,0 +1,425 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+YELLOW='\e[33;1m'
+RED='\e[31;1m'
+GREEN='\e[32;1m'
+ENDCOLOR='\e[0m'
+
+MODULE_NAME="tools"
+RELATIVE_QUAL_LOG_PATH="src/test/resources/spark-events-qualification"
+RELATIVE_PROF_LOG_PATH="src/test/resources/spark-events-profiling"
+RELATIVE_QUAL_REF_PATH="src/test/resources/QualificationExpectations"
+
+PROJECT_ROOT=""
+MODULE_PATH=""
+PROF_LOG_DIR=""
+QUAL_LOG_DIR=""
+QUAL_REF_DIR=""
+JVM_DEFAULT_HEAP_SIZE="10g"
+
+# get the directory of the script
+WORK_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+BATCH_OUT_DIR="${WORK_DIR}/qualification-output"
+CSV_OUT_DIR="${BATCH_OUT_DIR}/csv"
+RUNS_OUT_DIR="${BATCH_OUT_DIR}/runs-output"
+
+java_cp=""
+rapids_jar_home=""
+script_out_dir=""
+qual_events_dir=""
+qual_expectations_dir=""
+
+failed_tests=0
+executed_tests=0
+
+
+heap_size=$JVM_DEFAULT_HEAP_SIZE
+declare -A qualification_path_map
+
+show_help()
+{
+   # Display Help
+   log_msg
+   log_msg "Run the Qualification tool to generate CSV files to update/replace Qualification expectations set."
+   log_msg "The script requires the following:"
+   log_msg "  1- A static var \$qualification_path_map <string, array<string>> that maps between"
+   log_msg "     unit-test name and the array of event logs to be processed for the test. This map"
+   log_msg "     needs to be updated as new test are added and/or log files of existing tests are"
+   log_msg "     modified."
+   log_msg "  2- \$qual_expectations_dir: A directory that has the expected csv files."
+   log_msg "     For each key K in \$qualification_path_map, a csv file should exist"
+   log_msg "     \${qual_expectations_dir}/\$k.csv"
+   log_msg "  3- \$qual_events_dir: A directory that has the event logs to be processed."
+   log_msg
+   log_msg "The script iterates on the unit tests. For each key \"K\" in \$qualification_path_map:"
+   log_msg "  1- It runs the Qualification tool given the specified event logs."
+   log_msg "     The argument qual_out_dir=\${out-dir}/qualification-output/runs-output/\$K"
+   log_msg "     QualificationMain --output-directory \${qual_out_dir} \\"
+   log_msg "                        \${qualification_path_map[\$K]}"
+   log_msg "  2- It copies the rapids_4_spark_qualification_output.csv from \$qual_out_dir/"
+   log_msg "     to a new folder \${out-dir}/qualification-output/csv/\$K.csv"
+   log_msg "  3- Logs the result of \"diff\" between the two CSV files"
+   log_msg "     diff \${qual_expectations_dir}/\$K.csv \${out-dir}/qualification-output/csv/\$K.csv"
+   log_msg
+   log_msg "Finally, the script logs the result of \"diff\" for two directories:"
```

Comment:
this is same statement as 3- above

Reply:
I clarified the difference.
"3-" is diff between two CSV files.

The following step is to compare the entire two directories. I found that this step is necessary in case the Map is missing keys. Comparing the final directory is a second level to validate that the Map is not broken.